demo/runtime/src/runtime/council.rs

```diff
@@ -0,0 +1,1392 @@
+				v.push(signed.clone());
+				v
+			});
+		}
+		storage::put(&signed.to_keyed_vec(APPROVALS_OF), votes);
+		storage::put(&signed.to_keyed_vec(LAST_ACTIVE_OF), &index);
+	}
+
+	/// Remove a voter. For it not to be a bond-consuming no-op, all approved candidate indices
+	/// must now be either unregistered or registered to a candidate that registered the slot after
+	/// the voter gave their last approval set.
+	///
+	/// May be called by anyone. Returns the voter deposit to `signed`.
+	pub fn kill_inactive_voter(signed: &AccountId, signed_index: u32, who: &AccountId, who_index: u32, assumed_vote_index: VoteIndex) {
+		assert!(!presentation_active());
+		assert!(voter_last_active(signed).is_some());
+		let last_active = voter_last_active(who).expect("target for inactivity cleanup must be active");
+		assert!(assumed_vote_index == vote_index());
+		assert!(last_active < assumed_vote_index);
+		let voters = voters();
+		let signed_index = signed_index as usize;
+		let who_index = who_index as usize;
+		assert!(signed_index < voters.len() && voters[signed_index] == *signed);
+		assert!(who_index < voters.len() && voters[who_index] == *who);
+
+		// will definitely kill one of signed or who now.
+
+		let valid = !approvals_of(who).iter()
+			.zip(candidates().iter())
+			.any(|(&appr, addr)|
+				appr &&
+				*addr != AccountId::default() &&
+				candidate_reg_info(addr)
+					.expect("all items in candidates list are registered").0 <= last_active);
+
+		remove_voter(
+			if valid { who } else { signed },
+			if valid { who_index } else { signed_index },
+			voters
+		);
+		if valid {
+			staking::internal::set_balance(signed, staking::balance(signed) + voting_bond());
+		}
+	}
+
+	/// Remove a voter. All votes are cancelled and the voter deposit is returned.
+	pub fn retract_voter(signed: &AccountId, index: u32) {
+		assert!(!presentation_active());
+		assert!(storage::exists(&signed.to_keyed_vec(LAST_ACTIVE_OF)));
+		let voters = voters();
+		let index = index as usize;
+		assert!(index < voters.len() && voters[index] == *signed);
+		remove_voter(signed, index, voters);
+		staking::internal::set_balance(signed, staking::balance(signed) + voting_bond());
+	}
+
+	/// Submit oneself for candidacy.
+	///
+	/// Account must have enough transferrable funds in it to pay the bond.
+	pub fn submit_candidacy(signed: &AccountId, slot: u32) {
+		assert!(!is_a_candidate(signed));
+		let b = staking::balance(signed);
+		let candidacy_bond = candidacy_bond();
+		assert!(b >= candidacy_bond);
+		assert!(staking::unlock_block(signed) == staking::LockStatus::Liquid);
+
+		let slot = slot as usize;
+		let count = storage::get_or_default::<u32>(CANDIDATE_COUNT) as usize;
+		let candidates: Vec<AccountId> = storage::get_or_default(CANDIDATES);
+		assert!(
+			(slot == count && count == candidates.len()) ||
+			(slot < candidates.len() && candidates[slot] == AccountId::default())
+		);
+
+		staking::internal::set_balance(signed, b - candidacy_bond);
```

Comment:
what if this balance is already staked?

Reply:
it's not ideal, but i don't think it's a huge issue. if a validator is planning on losing their stake in an attack on the network, they're unlikely to care to use their future-dead-stake in a (comparatively minor) compute DoS attempt in a vote.

Reply:
it can provide an easy protection against slashing by temporarily moving away stake before misbehaving. complicates reasoning about incentives since staked tokens are meant to be completely infungible

Reply:
that theoretical protection ain't great, true (though it's unlikely to be a practical problem - you'd almost certainly lose your stake anyway since the candidate would be dropped at the next tally).

Reply:
a subsequent PR can introduce the idea of lockable stake into the staking system so that it would provide no protection against getting slashed.